pandas/plotting/_core.py

```diff
@@ -597,103 +397,227 @@ def boxplot_frame_groupby(grouped, subplots=True, column=None, fontsize=None,
+        When using a secondary_y axis, automatically mark the column
+        labels with "(right)" in the legend
+    `**kwds` : keywords
+        Options to pass to matplotlib plotting method
 
-    The backend is imported lazily, as matplotlib is a soft dependency, and
-    pandas can be used without it being installed.
+    Returns
+    -------
+    :class:`matplotlib.axes.Axes` or numpy.ndarray of them
+        If the backend is not the default matplotlib one, the return value
+        will be the object returned by the backend.
+
+    Notes
+    -----
+    - See matplotlib documentation online for more on this subject
+    - If `kind` = 'bar' or 'barh', you can specify relative alignments
+      for bar plot layout by `position` keyword.
+      From 0 (left/bottom-end) to 1 (right/top-end). Default is 0.5
+      (center)
     """
-    backend_str = pandas.get_option('plotting.backend')
-    if backend_str == 'matplotlib':
-        backend_str = 'pandas.plotting._matplotlib'
-    return importlib.import_module(backend_str)
+    _common_kinds = ('line', 'bar', 'barh', 'kde', 'density', 'area', 'hist',
+                     'box')
+    _series_kinds = ('pie',)
+    _dataframe_kinds = ('scatter', 'hexbin')
+    _kind_aliases = {'density': 'kde'}
+    _all_kinds = _common_kinds + _series_kinds + _dataframe_kinds
 
+    def __init__(self, data):
+        self._parent = data
 
-def _plot_classes():
-    plot_backend = _get_plot_backend()
-    # TODO restore type annotations if we create a base class for plot classes
-    # (a parent of MPLPlot, and classes of other backends)
-    classes = [plot_backend.LinePlot, plot_backend.BarPlot,
-               plot_backend.BarhPlot, plot_backend.AreaPlot,
-               plot_backend.HistPlot, plot_backend.BoxPlot,
-               plot_backend.ScatterPlot, plot_backend.HexBinPlot,
-               plot_backend.KdePlot, plot_backend.PiePlot]
-    return {class_._kind: class_ for class_ in classes}
-
-
-def _plot(data, x=None, y=None, subplots=False,
-          ax=None, kind='line', **kwds):
-    kind = _get_standard_kind(kind.lower().strip())
-    if kind in _all_kinds:
-        klass = _plot_classes()[kind]
-    else:
-        raise ValueError("%r is not a valid plot kind" % kind)
-
-    if kind in _dataframe_kinds:
-        if isinstance(data, ABCDataFrame):
-            plot_obj = klass(data, x=x, y=y, subplots=subplots, ax=ax,
-                             kind=kind, **kwds)
+    @staticmethod
+    def _get_call_args(data, args, kwargs):
+        """
+        We used to have different accessors for Series and DataFrame. Their
+        signatures were different:
+
+        - SeriesPlotMethods.__call__(kind, ..., **kwargs)
+        - DataFramePlotMethods.__call__(x, y, kind, ..., **kwargs)
+
+        This function makes this unified `__call__` method compatible with both
+        """
+        if args and isinstance(data, ABCSeries):
+            # TODO raise warning here, positional arguments shouldn't be
```

Comment:
can you add this raise? do we test this? alt if you'd make an issue an reference it here, we don't want to have TODOs w/o an issue.